Sweeney Hubspot Integration and Email Attachments

Logan-Ward opened the pull request:
Email api still needs to be tested

sscaff1 commented:
Check out the changes I made in #23. Couple of tips:

* The website is using JSS. If you're not familiar with JSS, don't just make an assumption that you can use CSS. That would be frowned upon in the job world. Ask a question first. You have to work with the constraints of the current system - not just add more crap to it.
* The form you made is not accessible and it's not implemented correctly. I encourage you to take a look at Kent Dobb's Epic React course it can really help. Formik which is what I ended up using is great, but not entirely necessarily. It just reduces the amount of code I ended up writing.
* The website is using Recaptcha v3. So it's actually much simpler that way.
* I talked to Devon. Sending attachments is expensive. So we won't be sending that via emails. 
* The site wasn't building for you because the package-lock was messed up. Just FYI and something that will help you in your career. Avoid the use of `npm install`, instead prefer `npm ci`. `npm ci` respects the package-lock and doesn't modify it (which is important). I use node/npm/react/JS/TS every single day of my life and I rarely use npm install. 
* `path` and `fs` are available through node. There is no reason to install them - I actually don't even know what the npm packages are - probably spam. 
* When you were reading the files on the backend, the only reason it was working for you locally was because you were testing it from your own computer. You can't read file paths that don't exist on the server (which is what it was going to attempt to do). 

Anyway, let me know if you have any questions. Good luck with your career.

Logan-Ward commented:
If you want more details or have any questions regarding my response, let me know. I had a longer write-up of everything you got wrong here, but I shortened it to this for readability's sake. 

> * The website is using JSS. If you're not familiar with JSS, don't just make an assumption that you can use CSS. That would be frowned upon in the job world. Ask a question first. You have to work with the constraints of the current system - not just add more crap to it.

The CSS was from the CoConstruct **_stylesheet_** you linked at the top of contact-us. I did not "just add more crap". Also, you did not use JSS in contact-us, you used CSS and inline styles, so I did the same. Do not assume I am making changes randomly. **_Ask a question first_**.

> * The form you made is not accessible and it's not implemented correctly. I encourage you to take a look at Kent Dobb's Epic React course it can really help. Formik which is what I ended up using is great, but not entirely necessarily. It just reduces the amount of code I ended up writing.

The form is accessible and implemented fine; it is imitating the original form that was managed through the CoConstruct JavaScript file at the top of contact-us. I know perfectly well how to make forms that follow better conventions. I just wanted to make as few changes as possible to your original implementation and not add multiple new dependencies **_like you did_**.

> * `path` and `fs` are available through node. There is no reason to install them - I actually don't even know what the npm packages are - probably spam.

I checked the npm packages; they are basically just stubs so that any project that "installs" them doesn't have any issues.

> * When you were reading the files on the backend, the only reason it was working for you locally was because you were testing it from your own computer. You can't read file paths that don't exist on the server (which is what it was going to attempt to do).

This is just **_completely wrong_** for many reasons. I recommend reading up on formidable's documentation. To assuage your concerns until then, I tested my code by deploying it to Vercel myself and it worked just fine, even when I went to the deployed link on my phone and submitted the form with an image attachment. Your version **_did not work at all_** as there were several errors.

Your code introduced **_multiple breaking runtime errors_** for the HubSpot integration. I have fixed all of them and cleaned up some front-end issues you made as well. The pull request will be up soon. 

Good luck with your career.
